### PR TITLE
feat: implement packet capture API

### PR DESCRIPTION
**Pull request by smira**
This uses the `go-packet` library with native bindings for the packet
capture (without `libpcap`). This is not the most performant way, but it
allows us to avoid CGo.

There is a problem with converting network filter expressions (like
`tcp port 3222`) into BPF instructions, it's only available in C
libraries, but there's a workaround with `tcpdump`.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 1]
```diff
@@ -2044,6 +2048,97 @@ func (s *Server) GenerateClientConfiguration(ctx context.Context, in *machine.Ge
 	return reply, nil
 }
 
+// PacketCapture performs packet capture and streams the pcap file.
+//
+//nolint:gocyclo
+func (s *Server) PacketCapture(in *machine.PacketCaptureRequest, srv machine.MachineService_PacketCaptureServer) error {
+
+	log.Printf("in PacketCapture")
```
Comment (frezbo) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:2056]: debug

Comment (smira) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:2056]: yeah, lots of debug still :) it's heavy WIP

File: cmd/talosctl/cmd/talos/pcap.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+ BPF filter can be applied, but it has to compiled to BPF instructions first using tcpdump.
+ Correct link type should be specified for the tcpdump: EN10MB for Ethernet links and RAW
+ for e.g. Wireguard tunnels:
+
+   talosctl pcap -i eth0 --bpf-filter "$(tcpdump -dd -y EN10MB 'tcp and dst port 80')"
+
+   talosctl pcap -i kubespan --bpf-filter "$(tcpdump -dd -y RAW 'port 50000')"
+
+As packet capture is transmitted over the network, it is recommended to filter out the Talos API traffic,
+e.g. by excluding packets with the port 50000.
```
Comment (frezbo) [inline cmd/talosctl/cmd/talos/pcap.go:64]: probably good to have an example that filters traffic on port 50000

**Review by frezbo — APPROVED** [review]
:cool:

File: cmd/talosctl/cmd/talos/pcap.go (hunk 2 above)
Comment (smira) [reply cmd/talosctl/cmd/talos/pcap.go:64]: also there might be some fun if the request is forwarded between apids, but still port 50000.

Comment (smira) [reply cmd/talosctl/cmd/talos/pcap.go:64]: I'm torn on that... We can only build a filter with tcpdump on client side. We need to know capture data type (-y). Client should have tcpdump installed.

**Pull request merged**